exporter/awsemfexporter/metric_translator.go

```diff
@@ -140,48 +141,116 @@ func TranslateOtToCWMetric(rm *pdata.ResourceMetrics, config *Config) ([]*CWMetr
 	if len(namespace) == 0 {
 		namespace = defaultNameSpace
 	}
+	return namespace
+}
 
-	ilms := rm.InstrumentationLibraryMetrics()
-	for j := 0; j < ilms.Len(); j++ {
-		ilm := ilms.At(j)
-		if ilm.IsNil() {
-			continue
-		}
-		if ilm.InstrumentationLibrary().IsNil() {
-			instrumentationLibName = noInstrumentationLibraryName
-		} else {
-			instrumentationLibName = ilm.InstrumentationLibrary().Name()
-		}
+// TranslateOtToGroupedMetric converts OT metrics to GroupedMetric format
+func TranslateOtToGroupedMetric(metric pdata.Metrics, groupedMetricMap map[string]*GroupedMetric) {
+	totalDroppedMetrics := 0
 
-		metrics := ilm.Metrics()
-		for k := 0; k < metrics.Len(); k++ {
-			metric := metrics.At(k)
-			if metric.IsNil() {
-				totalDroppedMetrics++
+	var dps DataPoints
+	rms := metric.ResourceMetrics()
+	for i := 0; i < rms.Len(); i++ {
+		rm := rms.At(i)
+		namespace := GetNamespaceFromMetric(&rm)
+		ilms := rm.InstrumentationLibraryMetrics()
+		for j := 0; j < ilms.Len(); j++ {
+			ilm := ilms.At(j)
+			if ilm.IsNil() {
 				continue
 			}
-			cwMetrics := getCWMetrics(&metric, namespace, instrumentationLibName, config)
-			cwMetricList = append(cwMetricList, cwMetrics...)
+
+			metrics := ilm.Metrics()
+			for k := 0; k < metrics.Len(); k++ {
+				metric := metrics.At(k)
+				if metric.IsNil() {
+					totalDroppedMetrics++
+					continue
+				}
+
+				switch metric.DataType() {
+					case pdata.MetricDataTypeIntGauge:
+						dps = IntDataPointSlice{metric.IntGauge().DataPoints()}
+					case pdata.MetricDataTypeDoubleGauge:
+						dps = DoubleDataPointSlice{metric.DoubleGauge().DataPoints()}
+					case pdata.MetricDataTypeIntSum:
+						dps = IntDataPointSlice{metric.IntSum().DataPoints()}
+					case pdata.MetricDataTypeDoubleSum:
+						dps = DoubleDataPointSlice{metric.DoubleSum().DataPoints()}
+					case pdata.MetricDataTypeDoubleHistogram:
+						dps = DoubleHistogramDataPointSlice{metric.DoubleHistogram().DataPoints()}
+				}
+
+				if dps.Len() == 0 {
+					return
+				}
+
+				for m := 0; m < dps.Len(); m++ {
+					dp := dps.At(m)
+					if dp.IsNil() {
+						continue
+					}
+
+					labelSlice := make([]string, 0, dp.LabelsMap().Len())
+					dp.LabelsMap().ForEach(func(k string, v string) {
+						labelSlice = append(labelSlice, k)
+					})
+					sort.Strings(labelSlice)
+					key := strings.Join(labelSlice, "")
```

Comment:
Need other field key and value to be batching key. i.e `Namespace`, `instrumentationLibName`, `Timestamp`

Reply:
I added `Namespace` value as part of the batching key